src/index.tsx

```diff
@@ -1,13 +1,36 @@
 import { createRoot } from 'react-dom/client';
-import { HashRouter as Router } from 'react-router-dom';
+import {
+  Navigate,
+  Route,
+  HashRouter as Router,
+  Routes,
+} from 'react-router-dom';
 
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { App } from './App';
+import { GlobalContextProvider } from './store/GlobalContextProvider';
+import { HomePage } from './components/HomePage';
+import { PeoplePage } from './components/PeoplePage';
+import { PageNotFound } from './components/PageNotFound';
 
 createRoot(document.getElementById('root') as HTMLDivElement).render(
-  <Router>
-    <App />
-  </Router>,
+  <GlobalContextProvider>
+    <Router>
+      <Routes>
+        <Route path="/" element={<App />}>
+          <Route index element={<HomePage />} />
+
+          <Route path="home" element={<Navigate to={'/'} />} />
+
+          <Route path="people" element={<PeoplePage />}>
+            <Route path=":slug?" element={<PeoplePage />} />
```

Comment:
The nested `Route` for `PeoplePage` with `path=":slug?"` is redundant since `PeoplePage` is already handling the slug parameter internally. Consider removing the nested `Route` to simplify the routing structure.